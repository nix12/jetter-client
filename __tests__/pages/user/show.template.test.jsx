import React from 'react';
import { shallow } from 'enzyme';

import Template from '../../../templates/user/show.template';

describe('<Show /> template', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      id: 1,
      username: 'test'
    };

    wrapper = shallow(<Template {...props} />);
  });

  it('should render Show component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call initialProps and output servside props', async () => {
    const isServer = true;
    const req = {
      userId: 1,
      username: 'test'
    };

    props = await Template.getInitialProps({ req, isServer });
    wrapper = shallow(<Template {...props} />);

    expect(props).toEqual({ userId: 1, username: 'test' });
  });
});
