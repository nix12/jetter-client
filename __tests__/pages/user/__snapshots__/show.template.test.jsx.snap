// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Show /> template should render Show component 1`] = `ShallowWrapper {}`;
